CRM/MembershipExtras/BAO/MembershipPeriod.php

```diff
@@ -223,12 +231,7 @@ public static function getOrderedMembershipPeriods($membershipId) {
    *
    * @throws CRM_Core_Exception
    */
-  public static function updatePeriod($params) {
-    if (self::doesOverlapWithOtherActivePeriods($params)) {
-      throw new CRM_Core_Exception('The new end date of this membership period overlaps with
-       another activated membership period. please review your changes');
-    }
-
+  public static function updatePeriodAndMembership($params) {
```

Comment:
I still think updatPeriod is the right name instead of updatePeriodAndMembership, I feel it better to abstract the concept of period since its known that updating a period should also be reflected on its membership.